library/Icinga/Web/RememberMe.php

```diff
@@ -217,27 +220,44 @@ public function authenticate()
             $user->setDomain(Config::app()->get('authentication', 'default_domain'));
         }
 
-        $authenticated = $authChain->authenticate($user, $password);
-        if ($authenticated) {
-            $auth->setAuthenticated($user);
+        foreach (AesCrypt::METHODS as $method) {
+            $this->aesCrypt->setMethod($method);
+            try {
+                $password = $this->aesCrypt->decryptFromBase64($this->encryptedPassword);
+            } catch (RuntimeException $e) {
+                Logger::error($e);
+                continue;
+            }
+
+            if (empty($password)) {
+                continue;
+            }
+
+            $authenticated = $authChain->authenticate($user, $password);
+
+            if ($authenticated) {
+                $auth->setAuthenticated($user);
+                break;
+            }
         }
 
+
         return $authenticated;
     }
 
     /**
      * Persist the remember me information into the database
      *
-     * Any previous stored information is automatically removed.
+     * To remove any previous stored information, set the iv
      *
-     * @param string|null $iv
+     * @param string|null $iv To remove a specific iv record from the database
      *
      * @return $this
      */
     public function persist($iv = null)
     {
         if ($iv) {
-            $this->remove(bin2hex($iv));
+            self::remove(bin2hex($iv));
         }
 
         $this->getDb()->insert(static::TABLE, [
```

Comment:
Just an observation: This now *may* establish a connection to the database twice. (`remove` uses `getDb()` as well but maintains a different instance of `ipl\Sql\Connection`)